pkg/domain/podman/router.go

```diff
@@ -106,13 +106,17 @@ func (r *RouterEntityManager) QueryAllRouters() ([]qdr.Router, error) {
 		var records []qdr.Record
 		err = json.Unmarshal([]byte(rJson), &records)
 		if err != nil {
+			// routerToQuery no longer exists
+			if routerToQuery != "" {
+				continue
+			}
 			return nil, fmt.Errorf("error decoding router info from %s - %w", routerToQuery, err)
 		}
 		router.Site = qdr.GetSiteMetadata(records[0].AsString("metadata"))
 
 		// retrieving connections
 		conns, err := r.QueryConnections(routerToQuery, router.Edge)
-		if err != nil {
+		if err != nil && !strings.Contains(err.Error(), "invalid character") {
```

Comment:
It is not at all obvious why 'invalid character' is 'expected' here. The issue I think is that any unusual output from the exec will fail to be parsed as json, most likely with this same error. One option that I think would at least be clearer is just to ignore the error here. Alternatively, to skip only expected errors, the QueryConnections() call could include the exec output in the error it returns, That way you can query for the actual expected error here (e.g. no such router or whatever the skmanage command reports).

Reply:
My inclination would be to return the output from exec in the error if failing to unmarshal, and in this specific case check for 'SendException: RELEASED' in the error and ignore it if so (ideally with a comment explaining that this most likely means the routerId did not exist anymore).

Ignoring the error entirely is not terrible either though.